aion/web/models/question.ex

```diff
@@ -37,10 +37,26 @@ defmodule Aion.Question do
   end
 
   @spec get_random_question(integer) :: Question.t
-  def get_random_question(category_id) do
-    query = from q in Question, where: q.subject_id == ^category_id
-    query
-    |> Repo.all()
-    |> Enum.random()
+  def get_random_question(room_id) do
+    query = Repo.query("
+      SELECT content, image_name, q.id FROM questions AS q
+      JOIN room_categories AS rc ON rc.category_id = q.category_id
+      JOIN rooms AS r ON r.id = rc.room_id
+      WHERE room_id = $1::integer
+      ORDER BY RANDOM()
+      LIMIT 1;
+    ", [String.to_integer(room_id)])
+
+    case query do
+      {:ok, result} ->
+        case result.rows do
```

Comment:
Did you run credo? Maybe credo doesn't check it but according to Elixir styleguide there shouldn't be more than 1-level depth in case statements.

Reply:
I did, how would you rewrite it not to have nested case? `result` has a lot of properties, not only `rows` so deconstructing this would not look so good.

Reply:
Nvm, I've managed.

Reply:
The recommended approach is to extract logic to a separate function